### PR TITLE
Feature: Date/datetime year range option

### LINKED ISSUE #5442: Feature: Change year range for date fields.
As the title says. Currently it only allows the default while this is an option for the jQuery datepicker.

Related: https://wordpress.org/support/topic/date-picker-years-range/

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes #5442 

## Description

Adds a new option to change the year range for date/datetime fields.

## ChangeLog

* Enhancement: Add year range option to date & datetime fields.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by JoryHogeveen** [comment]
This will need to be manually merged with #5451

File: classes/fields/date.php [hunk 1]
```diff
@@ -111,6 +111,15 @@ public function options() {
 				),
 				'dependency' => true,
 			),
+			static::$type . '_year_range'      => array(
+				'label'   => __( 'Year range', 'pods' ),
+				'default' => '',
+				'type'    => 'text',
+				'help'    => sprintf(
```
Comment (sc0ttkclark) [inline classes/fields/date.php]: Can we show an example of at least 1-2 options in this help text? This isn't super helpful if we just have a link here.

File: classes/fields/datetime.php [hunk 2]
```diff
@@ -181,6 +181,15 @@ public function options() {
 					'hh_mm_ss' => date_i18n( 'H:i:s' ),
 				),
 			),
+			static::$type . '_year_range'      => array(
+				'label'   => __( 'Year range', 'pods' ),
+				'default' => '',
+				'type'    => 'text',
+				'help'    => sprintf(
```
Comment (sc0ttkclark) [inline classes/fields/datetime.php]: Can we show an example of at least 1-2 options in this help text? This isn't super helpful if we just have a link here.

File: ui/fields/date.php [hunk 3]
```diff
@@ -33,6 +33,11 @@
 	'firstDay'    => (int) get_option( 'start_of_week', 0 ),
 );
 
+$year_range = pods_v( $form_field_type . '_year_range', $options, "" );
```
Comment (sc0ttkclark) [inline ui/fields/date.php]:
```suggestion
$year_range = pods_v( $form_field_type . '_year_range', $options, '' );
```

File: ui/fields/datetime.php [hunk 4]
```diff
@@ -38,6 +38,11 @@
 	'firstDay'    => (int) get_option( 'start_of_week', 0 ),
 );
 
+$year_range = pods_v( $form_field_type . '_year_range', $options, "" );
```
Comment (sc0ttkclark) [inline ui/fields/datetime.php]:
```suggestion
$year_range = pods_v( $form_field_type . '_year_range', $options, '' );
```

**Comment by sc0ttkclark** [comment]
I like how this is freeform where you can put lots of different formats that it supports.

I also like the idea of having a few fields too:

* Year Range (pick: Any year, Between years, Custom)
* Start year (if Between)
* End year (if Between)
* Custom range (accepts any datepicker year range format) *this is what you built so far*

Do you think it would be beneficial for us to introduce this feature with what I described above instead, to make it easier for folks who don't want to get into the weeds of the customized formats?

**Comment by JoryHogeveen** [comment]
@sc0ttkclark I think these extra UI options sounds nice but freeform format is probably just as easy. 
The Start/End year format is quite easy and we can indeed add an example in the help text. This will probably cover 90% of use cases.

Anyhow, for the next release I'd say lets stick with freeform so lets make that a new issue.

File: classes/fields/date.php [hunk 5]
```diff
@@ -111,6 +111,20 @@ public function options() {
 				),
 				'dependency' => true,
 			),
+			static::$type . '_year_range'      => array(
```
Comment (sc0ttkclark) [inline classes/fields/date.php]:
```suggestion
			static::$type . '_year_range_custom' => array(
```

File: classes/fields/datetime.php [hunk 6]
```diff
@@ -181,6 +181,20 @@ public function options() {
 					'hh_mm_ss' => date_i18n( 'H:i:s' ),
 				),
 			),
+			static::$type . '_year_range'      => array(
```
Comment (sc0ttkclark) [inline classes/fields/datetime.php]:
```suggestion
			static::$type . '_year_range_custom' => array(
```

File: ui/fields/date.php [hunk 7]
```diff
@@ -33,6 +33,11 @@
 	'firstDay'    => (int) get_option( 'start_of_week', 0 ),
 );
 
+$year_range = pods_v( $form_field_type . '_year_range', $options, '' );
```
Comment (sc0ttkclark) [inline ui/fields/date.php]:
```suggestion
$year_range = pods_v( $form_field_type . '_year_range_custom', $options, '' );
```

File: ui/fields/datetime.php [hunk 8]
```diff
@@ -38,6 +38,11 @@
 	'firstDay'    => (int) get_option( 'start_of_week', 0 ),
 );
 
+$year_range = pods_v( $form_field_type . '_year_range', $options, '' );
```
Comment (sc0ttkclark) [inline ui/fields/datetime.php]:
```suggestion
$year_range = pods_v( $form_field_type . '_year_range_custom', $options, '' );
```

**Pull request merged**